collision_detection/include/moveit/collision_detection/collision_common.h

```diff
@@ -130,6 +131,25 @@ namespace collision_detection
     }
   };
 
+  /** \brief Representation of a detailed distance */
+  struct DetailedDistance
+  {
+    DetailedDistance(std::string nearest_object, fcl::DistanceResult dist_result)
+    {
+      this->distance = dist_result.min_distance;
+      this->nearest_object = nearest_object;
+      this->nearest_points.first = Eigen::Vector3d(dist_result.nearest_points[0].data.vs);
```

Comment:
FCL distance check returns the [fcl::DistanceResult](http://gamma.cs.unc.edu/FCL/fcl_docs/webpage/generated/structfcl_1_1DistanceResult.html#0334911d9647bbb838c13bc7abd08d45) structure when checking two objects. The nearest _points is always of size 2. The first point is associated to object 1 and the second point is associated to object 2, representing the closest point between the two objects.